PotreeConverterPartitioned/potreeconverterpartitioned.py

```diff
@@ -0,0 +1,533 @@
+                        sbatchScript.write("#SBATCH " + parameters + "\n\n")
+                        sbatchScript.write(self.programPath +  " " + self.programOptions + " -o " + self.tmpOutputDir + " --header-dir " + self.lazHeadersDir)
+                    programCommand = shutil.which("sbatch") + " " + self.tmpDir + "/sbatchScript.sh"
+                    self.scheduler = SbatchScheduler(programCommand, self.programName)
+                elif config["SCHEDULER"]["Type"] == "pbs":
+                    parameters = config["SCHEDULER"]["Parameters"]
+                    with open(self.tmpDir + "/qsubScript.sh", "w") as qsubScript:
+                        qsubScript.write("#!/bin/sh\n\n")
+                        qsubScript.write("#PBS " + parameters + "\n\n")
+                        qsubScript.write(self.programPath + " " + self.programOptions + " -o " + self.tmpOutputDir + " --header-dir " + self.lazHeadersDir)
+                    programCommand = shutil.which("qsub") + " " + self.tmpDir + "/qsubScript.sh"
+                    self.scheduler = QsubScheduler(programCommand, self.programName)
+                elif config["SCHEDULER"]["Type"] == "local":
+                    programCommand = self.programPath + " " + self.programOptions + " -o " + self.tmpOutputDir + " --header-dir " + self.lazHeadersDir
+                    self.scheduler = LocalScheduler(programCommand, self.programName)
+                else:
+                    LoggingWrapper.error("Scheduler type not recognized. Must be one of sbatch, pbs or local")
+                    exit(1)
+            else:
+                LoggingWrapper.error("Scheduler type not found in SCHEDULER section")
+                exit(1)
+        else:
+            LoggingWrapper.error("SCHEDULER section not found in config file")
+            exit(1)
+
+
+    def bordered(self, text):
+        ''' This function is used to print a bordered text '''
+        lines = text.split("\n")
+        width = max(len(s) for s in lines)
+        res = ['=' * width]
+        for s in lines:
+            res.append('||' + (s + ' ' * width)[:width] + '||')
+        res.append('=' * width)
+        return '\n'.join(res)
+
+    def catFiles(self, filesToCat, destinationFile, batchCopier):
+        ''' This function is used to concatenate the output files of the PotreeConverterMPI program after processing a partition
+        :param filesToCat: The list of files to concatenate
+        :param destinationFile: The destination file
+        :param batchCopier: The copier object
+        '''
+
+        batchCopier.concatFiles(filesToCat, destinationFile)
+        batchCopier.removeFiles(filesToCat)
+
+
+    def testBatchDone(self, batchCopier, signalDir, state):
+        ''' This function is used to test if a batch/partition is done processing. If a batch is done processing, it will be removed from the batch dictionary and the corresponding signal files will be removed
+        :param batchCopier: The copier object
```

Comment:
These functions would be more readily understandable if a file naming schema (also for intermediate files such as partition or batch sub-files) were provided either in the preamble or as a separate readme. This would  support both the python wrappers as well as the C++ MPI code

Reply:
True